### ISSUE TITLE
fast decoding error when running t2t_decoder

### ISSUE THREAD

**Issue by fhieber** [tensorflow/tensor2tensor#397]
I am trying to decode a custom problem using standard transformer hyper-parameters.
Since 1.2.6 I am receiving the following error when running t2t-decoder:
```
...
INFO:tensorflow:decode_hp.batch_size not specified; default=32
INFO:tensorflow:Performing decoding from a file.
INFO:tensorflow:Getting sorted inputs
INFO:tensorflow: batch 94
INFO:tensorflow:Decoding batch 0
WARNING:tensorflow:Input graph does not use tf.data.Dataset or contain a QueueRunner. That means predict yields forever. This is probably a mistake.
WARNING:tensorflow:Schedule=train_and_evaluate. Assuming that training is running on a single machine.
INFO:tensorflow:datashard_devices: ['gpu:0']
INFO:tensorflow:caching_devices: None
INFO:tensorflow:Beam Decoding with beam size 5
Traceback (most recent call last):
  File "/usr/local/bin/t2t-decoder", line 103, in <module>
    tf.app.run()
  File "/usr/lib/python3.4/dist-packages/tensorflow/python/platform/app.py", line 48, in run
    _sys.exit(main(_sys.argv[:1] + flags_passthrough))
  File "/usr/local/bin/t2t-decoder", line 92, in main
    FLAGS.decode_to_file)
  File "/usr/local/lib/python3.4/site-packages/tensor2tensor/utils/decoding.py", line 235, in decode_from_file
    for result in result_iter:
  File "/usr/lib/python3.4/dist-packages/tensorflow/python/estimator/estimator.py", line 412, in predict
    features, None, model_fn_lib.ModeKeys.PREDICT, self.config)
  File "/usr/lib/python3.4/dist-packages/tensorflow/python/estimator/estimator.py", line 698, in _call_model_fn
    model_fn_results = self._model_fn(features=features, **kwargs)
  File "/usr/local/lib/python3.4/site-packages/tensor2tensor/utils/model_builder.py", line 349, in wrapping_model_fn
    return model_fn(model, features, mode, hparams, **kwargs)
  File "/usr/local/lib/python3.4/site-packages/tensor2tensor/utils/model_builder.py", line 164, in model_fn
    max_idx=len(hparams.problems) - 1)
  File "/usr/local/lib/python3.4/site-packages/tensor2tensor/utils/input_fn_builder.py", line 182, in cond_on_index
    return fn(cur_idx)
  File "/usr/local/lib/python3.4/site-packages/tensor2tensor/utils/model_builder.py", line 120, in nth_model
    decode_length=decode_hp.extra_length)
  File "/usr/local/lib/python3.4/site-packages/tensor2tensor/utils/t2t_model.py", line 220, in infer
    last_position_only, alpha)
  File "/usr/local/lib/python3.4/site-packages/tensor2tensor/models/transformer.py", line 183, in _beam_decode
    beam_size, top_beams, alpha)
  File "/usr/local/lib/python3.4/site-packages/tensor2tensor/models/transformer.py", line 214, in _fast_decode
    raise ValueError("Fast decoding only deals with the last positions!")
ValueError: Fast decoding only deals with the last positions!
```

I am not aware of setting any special decoding flags; this must have been introduced as a new default recently.
How can I get around this?

**Comment by vince62s** [comment]
From Gitter
vince62s @vince62s nov. 02 09:57
Team, following my previous post here: @vince62s
In the 1.2.6 details, here: Fast beam search decoding for the Transformer model (set --decode_hparams='use_last_position=True' with the t2t-decoder to use it)
you need to modify by use_last_position_only=true
(in the release description)
I would like to report that if you don't specify use_last_position_only=true , then it raises an error

but it may have been fixed in 1.2.7

**Comment by fhieber** [comment]
Thanks, this works! I can confirm that explicitly setting the parameter in 1.2.7 is still necessary.
